_posts/2019-08-18-view-encapsulation-w-angularze.md

```diff
@@ -0,0 +1,426 @@
+                        style="">
+                    <div aria-hidden="true"><span>Pobierz</span></div>
+                </label><label pseudo="-internal-media-controls-overflow-menu-list-item" role="menuitem" tabindex="0"
+                    aria-label=" Wycisz " class="animated-2" style="display: none;"><input type="button"
+                        pseudo="-webkit-media-controls-mute-button" tabindex="-1" aria-label="wyciszenie"
+                        style="display: none;">
+                    <div aria-hidden="true"><span>Wycisz</span></div>
+                </label><label pseudo="-internal-media-controls-overflow-menu-list-item" role="menuitem" tabindex="0"
+                    aria-label="odtwarzanie na urządzeniu zdalnym Przesyłaj " class="animated-1"
+                    style="display: none;"><input pseudo="-internal-media-controls-cast-button" type="button"
+                        aria-label="odtwarzanie na urządzeniu zdalnym" tabindex="-1" style="display: none;">
+                    <div aria-hidden="true"><span>Przesyłaj</span></div>
+                </label><label pseudo="-internal-media-controls-overflow-menu-list-item" role="menuitem" tabindex="0"
+                    aria-label="wyświetlanie menu napisów Napisy " class="animated-0" style="display: none;"><input
+                        aria-label="wyświetlanie menu napisów" type="button"
+                        pseudo="-webkit-media-controls-toggle-closed-captions-button" tabindex="-1" style="display: none;">
+                    <div aria-hidden="true"><span>Napisy</span></div>
+                </label><label pseudo="-internal-media-controls-overflow-menu-list-item" role="menuitem" tabindex="0"
+                    aria-label="włącz tryb obrazu w&nbsp;obrazie Obraz w&nbsp;obrazie " class="animated-0" style=""><input
+                        type="button" role="button" aria-label="włącz tryb obrazu w&nbsp;obrazie"
+                        pseudo="-internal-media-controls-picture-in-picture-button" tabindex="-1" style="">
+                    <div aria-hidden="true"><span>Obraz w&nbsp;obrazie</span></div>
+                </label></div>
+        </div>
+    <source src="movie.mp4" type="video/mp4">
+    <source src="movie.ogg" type="video/ogg">
+</video>
+```
+
+**Shadow DOM** ukrywa całą implementację pod prostym tagiem.  
+Dzięki temu style zaaplikowane do naszego elementu nie wpływają na inne elementy DOM'u.
+
+## View Encapsulation w Angularze
+
+Domyślnie Angular korzysta z własnego kapsułkowania styli (**ViewEncapsulation.Emulated**), ale udostępnia jeszcze 2 inne tryby kapsułkowania.  
+
+Aby zmienić domyślny tryb kapsułkowania, wystarczy dodać odpowiednią opcję w dekoratorze `@Component`, np.:  
+`encapsulation: ViewEncapsulation.ShadowDom`
+
+Omówimy je na przykładzie kodu z [**projektu demo.**](https://github.com/Michuu93/view-encapsulation-demo)
```

Comment:
1. Czy moglibyśmy ten projekt umieścić na githubie firmowym?
2. Przez to wytłuszczenie nie widać, że to link

Reply:
ad 1. już jest na moim, można zrobić forka ewentualnie - ale czy jest sens?
ad 2. poprawione